articles/remapping.md

```diff
@@ -0,0 +1,234 @@
+**Rationale:**
+This is part of the behavior of ROS 1 remapping, so including it will ease the transition to ROS 2.
+
+- A node uses names `/ns/bar` and `/ns/barista`
+- A rule is created to remap just `/ns/bar` to `/ns/foo`
+- The final names are `/ns/foo` and `/ns/barista`
+
+
+#### A user can remap a relative name to another name
+**Rationale:**
+This is part of the behavior of ROS 1 remapping.
+
+- A node is in namespace `/ns/` and uses name `bar`
+- A rule is created to remap `bar` to `foo`
+- The final name is `/ns/foo`
+
+
+#### A user can supply node specific remapping arguments via the command line
+This implies there is a way to uniquely identify a node in a process.
+
+**Rationale:**
+This is part of the behavior of ROS 1 remapping.
+
+
+#### A user can put a node into a specific namespace
+This implies there is a way to uniquely identify a node in a process.
+
+**Rationale:**
+ROS 1 has this feature using the environment variable `ROS_NAMESPACE` and the argument `__ns`
+
+- A system has multiple robots
+- A users wants the robots to not interfere with each other, so each robot is put into its own namespace
+- *TODO Global topics like /tf and /tf_static must be individually remapped in ROS 1. What to do about them?*
+
+
+#### A user can remap a name they see on a node after it has launched
+This implies remapping rules are applied in sequence.
+This also implies there is a way to supply remapping rules to a node besides the command line (services?).
+
+**Rationale:**
+A developer may want to make a running node subscribe to a different source of data to see how it behaves.
+It is less work for them to change the name they see rather than digging through the source code to find the original name.
+
+
+### Structure of a Remapping Rule
+Remapping rules have two parts:
+
+1. a part used to determine if the rule applies to a name
+2. a replacement for the names that are matched
+
+
+### Remapping Names in ROS 1
+Remapping is a feature that also exists in ROS 1.
+In ROS 1 remapping works by passing in [arguments](http://wiki.ros.org/Remapping%20Arguments) to each node.
+Client implementations also have APIs in code to pass remapping rules when the node is initialized.
+A remap rule consists of two names: one that should be replaced with another.
+
+ROS 1 remapping works on **Fully Qualified Names** (FQN).
+Both sides of a rule are [expanded to FQN](http://docs.ros.org/api/roscpp/html/namespaceros_1_1names.html#a377ff8fede7b95398fd5d1c5cd49246b).
+Before a name is remapped it is also [expanded to FQN](http://docs.ros.org/api/roscpp/html/namespaceros_1_1names.html#ab2eebaf734abfbdccb4122f8e24f547f).
+The name is remapped to the right side only if it exactly matches the left side of a rule.
+
+
+### Thoughts on ROS 2 implementation
+`rmw` is likely ignorant of remapping.
+The APIs offered by client implementations should remap names automatically to prevent creation of names that are impossible to remap.
```

Comment:
Maybe avoid the term `remap` here and replace it with `transform` to indicate that it is a different mechanism?